### PR TITLE
Epetra vector access operator

### PR DESCRIPTION
**Pull request by pcafrica**
This PR adresses - and hopefully solves - #16877.

### CODE REVIEW

**Review by bangerth — APPROVED** [review]
Thank you!

**Review by bangerth — CHANGES_REQUESTED** [review]
Let me retract the approval -- this is ok to merge if we can find a different place for the `using` declaration.

File: include/deal.II/lac/trilinos_epetra_vector.h [hunk 1]
```diff
@@ -50,6 +50,166 @@ namespace LinearAlgebra
...
+    {
+      /**
+       * Declare type for container size.
+       */
+      using size_type = dealii::types::global_dof_index;
+
+      /**
+       * Declare type for container value type.
+       */
+      using value_type = double;
```
[the remark this answers is not in the document]
Comment (pcafrica) [reply include/deal.II/lac/trilinos_epetra_vector.h:73]: Hi @bangerth, I'm not sure if I got your comment correctly.

I just followed the same pattern used in other similar classes (Tpetra Vector, TrilinosWrappers::MPI::Vector and so on).

What exactly do you mean by choosing a different namespace?

[the remark this answers is not in the document]
Comment (pcafrica) [reply include/deal.II/lac/trilinos_epetra_vector.h:73]: Do you just want to drop aliases from the `internal` namespace and leave them only in the main `Vector` class?

[the remark this answers is not in the document]
Comment (bangerth) [reply include/deal.II/lac/trilinos_epetra_vector.h:73]: What I meant is that having
```
  namespace EpetraWrappers {
    namespace internal {
      using value_type = double;
```
leaves a generically names type `value_type` in a namespace that is not clearly associated with the `Vector` class. It is perhaps true that *all* of the `EpetraWrappers` classes use `double` as their value type, but the point of the `value_type` you declare here is to support one specific class, and that should be made clear by either choosing a name that makes that obvious, or put it into a namespace that is clearly associated with the `Vector` class.

The ideal namespace, of course, is the class itself. So if there was a way to redirect all of the places where you use the `typdef` to the `Vector::value_type` type, that would be ideal.

Same for the other typedef (`size_type`).

[the remark this answers is not in the document]
Comment (pcafrica) [reply include/deal.II/lac/trilinos_epetra_vector.h]: Oh, I see your point. Would something like [this](c356f59de2a3754d64d8212bfa9d701cb5d04d51) work?

I'm noticing a similar pattern in other places (e.g. in `trilinos_vector.h` and `trilinos_tpetra_vector.h`). If my proposed solution is ok, I'll take care of propagating the modification there.

**Review by bangerth — APPROVED** [review]
Yes, this works. Thanks for working with me on this issue!

I'd be very happy to approve (separate) PRs if you wanted to fix this in other places as well!

**Comment by bangerth** [comment]
Would you mind fix-up'ing the indent commit into the commit it is supposed to fix?

**Comment by pcafrica** [comment]
> Would you mind fix-up'ing the indent commit into the commit it is supposed to fix?

Done!

**Comment by bangerth** [comment]
Thank you -- good to go!

**Pull request merged**